core/templates/dev/head/pages/exploration-editor-page/improvements-tab/services/improvements-display.service.ts

```diff
@@ -17,26 +17,49 @@
  * cards and data in the improvements tab of the exploration editor.
  */
 
-angular.module('oppia').factory('ImprovementsDisplayService', [function() {
-  var STATUS_CHOICES = {
-    open: {text: 'Open', cssClass: 'label label-info'},
-    fixed: {text: 'Fixed', cssClass: 'label label-default'},
-    ignored: {text: 'Ignored', cssClass: 'label label-default'},
-    compliment: {text: 'Compliment', cssClass: 'label label-success'},
-    not_actionable: {text: 'Not Actionable', cssClass: 'label label-default'},
-  };
+require('pages/exploration-editor-page/exploration-editor-page.constants.ts');
 
-  return {
-    isOpen: function(status) {
-      return status === 'open';
-    },
-    getStatusCssClass: function(status) {
-      return STATUS_CHOICES.hasOwnProperty(status) ?
-        STATUS_CHOICES[status].cssClass : 'label label-default';
-    },
-    getHumanReadableStatus: function(status) {
-      return STATUS_CHOICES.hasOwnProperty(status) ?
-        STATUS_CHOICES[status].text : '';
-    },
-  };
-}]);
+angular.module('oppia').factory('ImprovementsDisplayService', [
+  'STATUS_COMPLIMENT', 'STATUS_FIXED', 'STATUS_IGNORED',
+  'STATUS_NOT_ACTIONABLE', 'STATUS_OPEN',
+  function(
+      STATUS_COMPLIMENT, STATUS_FIXED, STATUS_IGNORED,
+      STATUS_NOT_ACTIONABLE, STATUS_OPEN) {
+    var STATUS_CHOICES = {
+      [STATUS_COMPLIMENT]: {
```

Comment:
Oh, cool -- I have never seen this construction before (for the keys of a dict being named constants). Do you have a reference for it? We might use this pattern elsewhere in the future.

Reply:
Had to look it up because I wanted to avoid using literals so bad :)

https://stackoverflow.com/questions/882727/is-there-a-way-to-use-variable-keys-in-a-javascript-object-literal

Reply:
Ah, that's great! I didn't know that was a thing.

Do you know what its browser support looks like?

Reply:
Looks like all but IE support it :( Should I get rid of it?

Reply:
Not sure you got the right link...

Hm, I'm not sure we need to support IE. I think we can keep it actually :)

Reply:
Whoops... here's the correct link: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Browser_compatibility